File: workout_app/lib/data/models/profil/profil_model.dart
import 'package:workout_app/data/entities/profil/profil_entity.dart';

class ProfilModel extends ProfilEntity {
  ProfilModel({
    required super.id,
    required super.name,
    required super.gender,
    required super.age,
    required super.weight,
    required super.height,
    required super.caloriesTarget,
    required super.carbsTarget,
    required super.proteinsTarget,
    required super.fatsTarget,
    required super.activityLevel,
    required super.goal,
  });

  factory ProfilModel.fromEntity(ProfilEntity entity) {
    return ProfilModel(
      id: entity.id,
      name: entity.name,
      gender: entity.gender,
      age: entity.age,
      weight: entity.weight,
      height: entity.height,
      caloriesTarget: entity.caloriesTarget,
      carbsTarget: entity.carbsTarget,
      proteinsTarget: entity.proteinsTarget,
      fatsTarget: entity.fatsTarget,
      activityLevel: entity.activityLevel,
      goal: entity.goal,
    );
  }

  factory ProfilModel.fromJson(Map<String, dynamic> json) {
    return ProfilModel(
      id: json['id'].toString(),
      name: json['name'].toString(),
      gender: json['gender'].toString(),
      age: (json['age'] as num).toInt(),
      weight: (json['weight'] as num).toDouble(),
      height: (json['height'] as num).toInt(),
      caloriesTarget: (json['caloriesTarget'] as num).toDouble(),
      carbsTarget: (json['carbsTarget'] as num).toDouble(),
      proteinsTarget: (json['proteinsTarget'] as num).toDouble(),
      fatsTarget: (json['fatsTarget'] as num).toDouble(),
      activityLevel: json['activityLevel'].toString(),
      goal: json['goal'].toString(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'gender': gender,
      'age': age,
      'weight': weight,
      'height': height,
      'caloriesTarget': caloriesTarget,
      'carbsTarget': carbsTarget,
      'proteinsTarget': proteinsTarget,
      'fatsTarget': fatsTarget,
      'activityLevel': activityLevel,
      'goal': goal,
    };
  }
}

File: backend/models/entities/profil_model.go
package entities

import (
	"time"

	"gorm.io/gorm"
)

type Profil struct {
	ID             string         `gorm:"primaryKey;type:uuid;default:gen_random_uuid()" json:"id"`
	Name           string         `gorm:"type:varchar(100);not null" json:"name"`
	Gender         string         `gorm:"type:varchar(10);not null" json:"gender"`
	Age            int            `gorm:"not null;check:age >= 10 AND age <= 120" json:"age"`
	Weight         float32        `gorm:"not null;check:weight >= 30 AND weight <= 200" json:"weight"`
	Height         int            `gorm:"not null;check:height >= 100 AND height <= 220" json:"height"`
	CaloriesTarget float64        `gorm:"not null;check:calories_target >= 0" json:"caloriesTarget"`
	CarbsTarget    float64        `gorm:"not null;check:carbs_target >= 0" json:"carbsTarget"` // CarbsTarget → carbs_target
	ProteinsTarget float64        `gorm:"not null;check:proteins_target >= 0" json:"proteinsTarget"`
	FatsTarget     float64        `gorm:"not null;check:fats_target >= 0" json:"fatsTarget"`
	ActivityLevel  string         `gorm:"type:varchar(50)" json:"activityLevel"`
	Goal           string         `gorm:"type:varchar(50)" json:"goal"`
	CreatedAt      time.Time      `gorm:"autoCreateTime" json:"createdAt"`
	UpdatedAt      time.Time      `gorm:"autoUpdateTime" json:"updatedAt"`
	DeletedAt      gorm.DeletedAt `gorm:"index" json:"-"`
}
